### DISCUSSION TITLE
Light Slider for last switched Light

### DISCUSSION THREAD

**Discussion by Frankhag** [ExperienceLovelace/ha-floorplan#442] [General]
Is it possible to have a slider that always controls the last light switched on? any idea how to implement this?

**Comment by exetico** [comment]
Hi,

Last light switched on, from what source? Only switched on within your floorplan, or overall in Home Assistant?

If its the last one, its more a question about saving in Home Assistant somehow, not anything related to ha-floorplan.

**Reply by exetico** [reply]
@Frankhag Answer with comments to a reply; Not a fresh reply-thread.

Okay, I see the use-case.

Do you only have a single screen with the floorplan, or?

Are the light slider just a static image, or how did you make that part? Is it a floating **normal** ha-element, outside the ha-floorplan instance? :-)

**Reply by Frankhag** [reply]
Yes, at the moment, i have it with a normal ha-element. I have a group of lights and the last light of this group, that was switched on, i can dimm with the slider.


              - type: custom:config-template-card
                variables:
                  LAST_LIGHT: states['sensor.last_light_on'].state
                entities:
                  - ${LAST_LIGHT}
                card:
                  type: custom:slider-button-card
                  entity: ${LAST_LIGHT}
                  slider:
                    direction: left-right
                    background: none;
                    use_state_color: false
                    use_percentage_bg_opacity: false
                    show_track: true
                    toggle_on_click: false
                    force_square: false
                  show_name: false
                  show_state: false
                  compact: true
                  icon:
                    show: false
                    haptic: medium
                    use_state_color: true
                    tap_action:
                      action: none
                    icon: mdi:format-color-fill
                  action_button:
                    show: false
                    icon: mdi:power-standby
                    tap_action:
                      action: none
                  show_attribute: true
                  card_mod:
                    style: |
                      ha-card {
                        --ha-card-background: none;
                        box-shadow: 0px 2px 2px 0px rgba(10, 10, 10, 0.6) !important;
                        border-radius: 5px !important;
                      }
                style:
                  top: 28%
                  left: 78% 
                  width: 32%     
                  z-index: 5

and the Sensor for last light on:

      - name: "Last Light On"
        unique_id: last_light_on_001
        state: |
          {{ state_attr('light.floorplan', 'entity_id') | select('is_state','on')| expand | sort(attribute = 'last_changed', reverse=1 ) 
          | map(attribute='entity_id') | first }}
        attributes:
          count: "{{states.light | selectattr('state', 'eq', 'on') | list | count}}"
        availability: |-
          {{states.light | selectattr('state', 'eq', 'on') | list | count > 0 }}

**Reply by exetico** [reply]
If so, you've solved the question, right?

We don't have a slider-implementation in ha-floorplan.

**Comment by Frankhag** [comment]
![slider](https://github.com/ExperienceLovelace/ha-floorplan/assets/59791991/5435d017-4124-450a-a069-75ecbc2ec19f)

this is my floorplan. I want that the light slider controll the last light i switched on from floorplan.  when it work like i want, i can hide it and so, i have a hidden light slider always on the same position and with the last light i switched on.
